rita_common/src/tunnel_manager/id_callback.rs

```diff
@@ -28,37 +28,30 @@ impl IdentityCallback {
     }
 }
 
-impl Message for IdentityCallback {
-    type Result = Option<(Tunnel, bool)>;
-}
-
-// An attempt to contact a neighbor has succeeded or a neighbor has contacted us, either way
-// we need to allocate a tunnel for them and place it onto our local storage.  In the case
-// that a neighbor contacts us we don't have a port already allocated and we need to choose one
-// in the case that we have atempted to contact a neighbor we have already sent them a port that
-// we now must attach to their tunnel entry. If we also return a bool for if the tunnel already
-// exists
-impl Handler<IdentityCallback> for TunnelManager {
-    type Result = Option<(Tunnel, bool)>;
+/// An attempt to contact a neighbor has succeeded or a neighbor has contacted us, either way
+/// we need to allocate a tunnel for them and place it onto our local storage.  In the case
+/// that a neighbor contacts us we don't have a port already allocated and we need to choose one
+/// in the case that we have atempted to contact a neighbor we have already sent them a port that
+/// we now must attach to their tunnel entry. If we also return a bool for if the tunnel already
+/// exists
+pub fn tm_identity_callback(msg: IdentityCallback) -> Option<(Tunnel, bool)> {
+    let tunnel_manager = &mut *TUNNEL_MANAGER.write().unwrap();
```

Comment:
can't you use let mut tunnel_manager = TUNNEL_MANAGER.write().unwrap() here? I find it a little cleaner

Reply:
fixed